pkg/build/strategies/dockerfile/dockerfile.go

```diff
@@ -139,13 +139,12 @@ func (builder *Dockerfile) CreateDockerfile(config *api.Config) error {
 			return errors.New("Image tag is missing for incremental build")
 		}
 		buffer.WriteString(fmt.Sprintf("FROM %s as cached\n", imageTag))
-		if len(config.AssembleUser) > 0 {
-			buffer.WriteString(fmt.Sprintf("USER %s\n", imageUser))
-		}
 		var artifactsScript string
 		if _, provided := providedScripts[constants.SaveArtifacts]; provided {
+			// Switch to root to COPY and chown content
 			glog.V(2).Infof("Override save-artifacts script is included in directory %q", builder.uploadScriptsDir)
 			buffer.WriteString("# Copying in override save-artifacts script\n")
+			buffer.WriteString("USER root\n")
```

Comment:
use `USER 0` instead?

Reply:
this should be fine i think, especially given our intent to remove it.